File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ESecondaryContainerLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESecondaryContainerLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl27
            AppTheme.STANDARD -> R.color.cstl27
            AppTheme.BLUE -> R.color.cbl27
            AppTheme.GREEN -> R.color.cgl27
            AppTheme.RED -> R.color.crl27
            AppTheme.NOTHING -> R.color.cnol27
        }
    }
}

File: lib/layer2_core/light/e_secondary_container_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ESecondaryContainerLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl27;
      case 'standard': return cstl27;
      case 'blue': return cbl27;
      case 'green': return cgl27;
      case 'red': return crl27;
      case 'nothing': return cnol27;
      default: return cstl27;
    }
  }
}
